### PR TITLE
exit if the user has requested a specific port

### PR DESCRIPTION
**Pull request by SvenDowideit**
Signed-off-by: Sven Dowideit <sven.dowideit@csiro.au>

for #945

to go with https://github.com/datproject/dat-node/pull/192

### CODE REVIEW

**Comment by SvenDowideit** [comment]
I tried to work out how to use the lib/exit.js code, but couldn't figure out a way that worked.

File: bin/cli.js [hunk 1]
```diff
@@ -20,7 +20,7 @@ var config = {
   defaults: [
     { name: 'dir', abbr: 'd', help: 'set the directory for Dat' },
     { name: 'logspeed', default: 400 },
-    { name: 'port', default: 3282, help: 'port to use for connections' },
+    { name: 'port', default: 0, help: 'port to use for connections (tries 3282 by default)' },
```
Comment (joehand) [inline bin/cli.js]: I think we'd be better removing this default. Otherwise `dat-node` won't try 3282 first.

Comment (joehand) [reply bin/cli.js]: Oh no it'll try 3828 first, nm. But still not sure we need this, just confuses things =).

**Comment by joehand** [comment]
> I tried to work out how to use the lib/exit.js code, but couldn't figure out a way that worked.

Cool, I can take a look at adding that.

**Comment by joehand** [comment]
Can you check out [these syntax changes](https://travis-ci.org/datproject/dat/jobs/350150764#L1043)

**Comment by SvenDowideit** [comment]
@joehand nice! thanks for the syntax pointer.

is there a test exception that can be configured for `import-progress.js`? to my naive reading, that seems like a false negative.

**Comment by joehand** [comment]
> is there a test exception that can be configured for import-progress.js? to my naive reading, that seems like a false negative.

ya not sure what was going on there. Made it one-liner to avoid indent error.

**Pull request merged**